ci-operator/step-registry/baremetalds/sno/hostname/baremetalds-sno-hostname-commands.sh

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Fetch packet basic configuration
+# shellcheck source=/dev/null
+source "${SHARED_DIR}/packet-conf.sh"
+
+ssh "${SSHOPTS[@]}" "root@${IP}" bash - << "EOF"
+
+hostname_script=$(cat << "IEOF"
+#!/usr/bin/env bash
+
+set -euoE pipefail
+
+export KUBECONFIG=/etc/kubernetes/static-pod-resources/kube-apiserver-certs/secrets/node-kubeconfigs/localhost.kubeconfig
+wait_for_api() {
+  echo "Waiting for API..."
+  until oc get clusterversion &> /dev/null
+  do
+    echo "Waiting for API..."
+    sleep 5
+  done
+  echo "API is available"
+}
+
+wait_approve_csr() {
+  local name=${1}
+
+  echo "Waiting for ${name} CSR..."
+  until oc get csr | grep -i "${name}" | grep -i "pending" &> /dev/null
+  do
+    echo "Waiting for ${name} CSR..."
+    sleep 5
+  done
+  echo "CSR ${name} is ready for approval"
+
+  echo "Approving all pending CSRs..."
+  oc get csr -o go-template='{{range .items}}{{if not .status}}{{.metadata.name}}{{"\n"}}{{end}}{{end}}' | xargs oc adm certificate approve
+}
+
+if [[ "$(hostname)" != "another-hostname" ]]
+then
+  systemctl stop kubelet.service
+  crictl rmp -fa
```

Comment:
```suggestion
  # The pods have to be removed completely rather than just stopped because a different hostname requires new pods to be created by kubelet
  crictl rmp -fa
```